api/envoy/extensions/filters/udp/dns_filter/v3alpha/dns_filter.proto

```diff
@@ -52,8 +53,25 @@ message DnsFilterConfig {
     google.protobuf.Duration resolver_timeout = 1 [(validate.rules).duration = {gte {seconds: 1}}];
 
     // DNS resolution configuration which includes the underlying dns resolver addresses and options.
+    // *dns_resolution_config* will be deprecated once
+    // :ref:'typed_dns_resolver_config <envoy_v3_api_field_extensions.filters.udp.dns_filter.v3alpha.DnsFilterConfig.ClientContextConfig.typed_dns_resolver_config>'
+    // is fully supported.
     config.core.v3.DnsResolutionConfig dns_resolution_config = 2;
 
+    // DNS resolver type configuration extension. This extension can be used to configure c-ares, apple,
+    // or any other DNS resolver types and the related parameters.
+    // For example, an object of :ref:`DnsResolutionConfig <envoy_v3_api_msg_config.core.v3.DnsResolutionConfig>`
```

Comment:
This packing is really there for the special case of maintaining existing behavior presumably? The more standard situation is to pack the extension config for c-ares or apple resolvers.

Reply:
Good catch. We are packing a cares or apple DNS resolver message, not DnsResolutionConfig message. PTAL the latest change.